framework/decode/dx12_dump_resources.cpp

```diff
@@ -540,20 +577,418 @@ void Dx12DumpResources::BeginRenderPass(
     }
 }
 
-bool MatchDescriptorCPUGPUHandle(size_t                                      replay_cpu_addr_begin,
-                                 size_t                                      replay_target_cpu_addr,
-                                 uint64_t                                    capture_gpu_addr_begin,
-                                 std::map<UINT, D3D12_GPU_DESCRIPTOR_HANDLE> captured_gpu_addrs)
+bool CaptureGPUAddrMatchDescriptorHeap(const D3D12_GPU_DESCRIPTOR_HANDLE capture_gpu_addr,
+                                       const D3D12DescriptorHeapInfo&    heap_info,
+                                       uint32_t&                         out_heap_index)
 {
-    for (auto gpu_addr : captured_gpu_addrs)
+    auto increment            = (*heap_info.capture_increments)[heap_info.descriptor_type];
+    auto capture_gpu_addr_end = heap_info.capture_gpu_addr_begin + heap_info.descriptor_count * increment;
+
+    bool is_match = true ? (heap_info.capture_gpu_addr_begin <= capture_gpu_addr.ptr &&
+                            capture_gpu_addr.ptr <= capture_gpu_addr_end)
```

Comment:
Should this be `capture_gpu_addr.ptr < capture_gpu_addr_end` instead of `capture_gpu_addr.ptr <= capture_gpu_addr_end`

Reply:
fixed